src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -252,49 +174,63 @@ def find_files_missing_header():
     parser = argparse.ArgumentParser()
     args = set_lint_args(parser)
 
-    # Get custom specified directories, copyright, template, and/or license
-    dirs = args.loc.split(",")
+    # Get committed files, the year, and custom copyright, template, and/or license arguments
+    files = args.files
     copyright = args.custom_copyright
     template = args.custom_template
     license = args.custom_license
-    changed_headers = False
-
-    # Get current year for license file
+    changed_headers = 0
     year = dt.today().year
 
+    # Get root directory of the git repository.
+    git_repo = git.Repo(os.getcwd(), search_parent_directories=True)
+    git_root = git_repo.git.rev_parse("--show-toplevel")
+
+    # Update file paths to be absolute paths with correct separators
+    for i in range(0, len(files)):
+        if "win" in sys.platform:
+            split_str = files[i].split("/")
+            files[i] = os.path.abspath(os.path.join(*split_str))
+        else:
+            files[i] = os.path.abspath(files[i])
+
     # Add header arguments to parser. Arguments are: copyright, license, contributor,
     # year, style, copyright-style, template, exclude-year, merge-copyrights, single-line,
     # multi-line, explicit-license, force-dot-license, recursive, no-replace,
     # skip-unrecognized, and skip-existing.
     header.add_arguments(parser)
 
-    # Check if required directories exist
-    for dir in dirs:
-        dir_exists = check_dir_exists(dir)
-        if not dir_exists:
-            # Previous check_dir_exists() function returned error because
-            # --loc's value does not exist... returning 2
-            return 2
-
-        proj = project.Project(rf"{dir}")
-        missing_headers = list_noncompliant_files(args, proj)
-
-        # If there are files missing headers, run REUSE and return 1
-        if missing_headers:
-            changed_headers = True
-            # Add missing license header to each file in the list
-            for file in missing_headers:
-                args = set_header_args(parser, dir, year, file, copyright, template)
-                # If adding license header for the first time
+    # Run REUSE on root of the repository
+    proj = project.Project(git_root)
+    missing_headers = list(list_noncompliant_files(args, proj))
+
+    def check_exists(changed_headers, i, j):
+        """Check if the committed file is missing its header."""
```

Comment:
This recursive method is a bit cumbersome... Can't you loop the files independently instead of implementing it recursively?

Reply:
I could, but I thought parsing through the files recursively would have a better time complexity, so it would be faster in cases where there are a lot of files in ``missing_headers`` or ``files``

Reply:
As long as there is a reason, that's fine. Thanks @klmcadams!